### PR TITLE
[MB-14732] DutyLocationSearchBoxComponent refactor

### PR DESCRIPTION
**Pull request by SirenaBorracha**
## Summary

This PR generalizes the `DutyLocationSearchBox` component for future use in the PPM closeout transportation office search. It does not implement this feature as the API query is still WIP. In pairing with Duncan, we determined that it to use the `transportationOfficeFactory` as that would have required the aforementioned API.

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

##### Terminal 2

Start the Go server locally.

```sh
make server_run
```

##### Terminal 3

Start the UI locally.

```sh
make client_run
```

</details>

### Additional steps

1. Open http://milmovelocal:3000/ and create a new service member profile
2. Progress through the flow up to the search for current duty station to confirm that the component is still working as expected

## Verification Steps for Author

These are to be checked by the author.

- [x] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [x] Request review from a member of a different team.
- [x] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/main/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#zero-downtime-migrations)
- [ ] Have been communicated to #g-database
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#secure-migrations)

## Screenshots
Screenshot 2023-01-11 at 12 54 38 PM

### CODE REVIEW

File: src/components/LocationSearchBox/LocationSearchBox.test.jsx [hunk 1]
```diff
@@ -16,98 +16,121 @@ const mockDutyLocations = [
   dutyLocationFactory(),
 ];
 
-const selectedDutyLocation = mockDutyLocations[2];
-const mockAddress = selectedDutyLocation.address;
+const selectedLocations = mockLocations[2];
```
Comment (mboren) [inline src/components/LocationSearchBox/LocationSearchBox.test.jsx:19]: teeny tiny nitpick: i think `selectedLocation` would make more sense than `selectedLocations` here.

**Review by mboren — APPROVED** [review]
I left one lil variable naming suggestion, but aside from that everything looks good to me.

File: src/components/LocationSearchBox/LocationSearchBox.jsx [hunk 2]
```diff
@@ -186,14 +185,14 @@ export const DutyLocationSearchBoxComponent = (props) => {
           loadOptions={loadOptions}
           onChange={selectOption}
           onInputChange={changeInputText}
-          placeholder={placeholder || 'Start typing a duty location...'}
-          value={hasDutyLocation ? value : null}
+          placeholder={placeholder || 'Start typing a location...'}
+          value={hasLocation ? value : null}
```
Comment (mboren) [inline src/components/LocationSearchBox/LocationSearchBox.jsx:189]: looks like this updated placeholder causes some happo diffs. If this isn't intentional, i guess you'll need to pass in the the placeholder prop when this component is used.

**Review by PDickeyTruss — APPROVED** [review]
Looking good to me, only issue I see is the one Marty noted around the placeholder text removing the word 'duty'. Approved pending a fix to the placeholder text by presumably either changing the default back or updating all usages to include the placeholder prop with the correct text.

**Review by dinabeigelman — COMMENTED** [review]
Took a look and here's what I found:

- I think [this link](https://output.circle-artifacts.com/output/job/aed29671-abbe-41dc-8647-5b9c6d85110b/artifacts/0/storybook/index.html?path=/story/components-location-search-box--duty-station-basic) to see the storybook component needs to be regenerated to show the placeholder copy changes you made.

 - What I have in my designs is "Start typing a closeout location..." for the office app and "Start typing a closeout location..." for the customer app "Start typing a closeout office..."

 - I think in your work it's "Start typing a closeout office..." which I believe is fine assuming we can change it once the work is implemented in the other tickets.

- I see two new sections created in storybook for "Transportation Location Basic" and Transportation Location With Value" but none for "Transportation Location with Value and Address" or "Transportation Location with Error Message" like we have for duty station. I don't know what the best practice here is but we will be using the "with value" variant in implementation and the error state should mimic how we treat duty station errors.

**Review by dinabeigelman — APPROVED** [review]
considering this is a component and configurable when placed in context of the apps i'm gonna go ahead and approve!

**Pull request merged**